pkg/envoy/xds/cache.go

```diff
@@ -254,3 +254,14 @@ func (c *Cache) GetResources(ctx context.Context, typeURL string, lastVersion *u
 	cacheLog.Debugf("returning %d resources out of %d requested", len(res.Resources), len(resourceNames))
 	return res, nil
 }
+
+// Lookup finds the resource corresponding to the specified typeURL and resourceName,
+// if available, and returns it. Otherwise, returns nil. If an error occurs while
+// fetching the resource, also returns the error.
+func (c *Cache) Lookup(typeURL string, resourceName string) (proto.Message, error) {
```

Comment:
Please add short unit tests for this method, for both the positive and negative cases.

Reply:
Added a couple of basic tests.